ml-agents/mlagents/trainers/demonstrations/local_demonstration_provider.py

```diff
@@ -0,0 +1,146 @@
+import os
+from typing import List
+import numpy as np
+
+
+from mlagents_envs.base_env import ActionTuple, BehaviorSpec, ActionSpec
+from mlagents_envs.communicator_objects.agent_info_action_pair_pb2 import (
+    AgentInfoActionPairProto,
+)
+from mlagents_envs.rpc_utils import steps_from_proto
+
+
+from mlagents.trainers.demonstrations.demonstration_provider import (
+    DemonstrationProvider,
+    DemonstrationExperience,
+    DemonstrationTrajectory,
+)
+from mlagents.trainers.demonstrations.demonstration_proto_utils import (
+    load_demonstration,
+)
+
+
+class LocalDemonstrationProvider(DemonstrationProvider):
+    def __init__(self, file_path: str):
+        super().__init__()
+
+        demo_paths = self._get_demo_files(file_path)
+        behavior_spec, info_action_pairs, = load_demonstration(demo_paths)
+        self._behavior_spec = behavior_spec
+        self._info_action_pairs = info_action_pairs
+
+    def get_behavior_spec(self) -> BehaviorSpec:
+        return self._behavior_spec
+
+    def pop_trajectories(self) -> List[DemonstrationTrajectory]:
```

Comment:
Need to add docstrings here. But the idea is that GAIL, etc could be converted to use pop_trajectories() directly. Then if we want DemonstrationProviders to be able to load new demonstrations on the fly, the logic can be kept in the DemonstrationProvider and the consumer doesn't need to know about it, it just gets a fresh batch of trajectories.